### ISSUE TITLE
Getting a cast exception in binding pattern

### ISSUE THREAD

**Issue by KavinduZoysa** [ballerina-platform/ballerina-lang#24473]
**Description:**
The following code has syntax errors and after parser recovers the errors, a cast exception is thrown in NodeTransformer

source code:

```
type Foo1 object {
    func1() {};
    func2(x) {
        x = x + 1
    };
}
```

error:
 ``` 
  java.lang.ClassCastException: io.ballerinalang.compiler.syntax.tree.MappingBindingPatternNode cannot be cast to io.ballerinalang.compiler.syntax.tree.CaptureBindingPatternNode
        at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:487)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:405)
        at io.ballerinalang.compiler.syntax.tree.ModuleVariableDeclarationNode.apply(ModuleVariableDeclarationNode.java:67)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:405)
        at io.ballerinalang.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:430)
        at org.wso2.ballerinalang.compiler.parser.Parser.generateCompilationUnitNew(Parser.java:143)
        at org.wso2.ballerinalang.compiler.parser.Parser.parseNew(Parser.java:116)
        at org.wso2.ballerinalang.compiler.PackageLoader.parse(PackageLoader.java:500)
        at org.wso2.ballerinalang.compiler.PackageLoader.loadEntryPackage(PackageLoader.java:380)
        at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:156)
        at org.wso2.ballerinalang.compiler.Compiler.compilePackage(Compiler.java:174)
        at org.wso2.ballerinalang.compiler.Compiler.compile(Compiler.java:91)
        at org.ballerinalang.test.util.BCompileUtil.compile(BCompileUtil.java:516)
        at org.ballerinalang.test.util.BCompileUtil.compile(BCompileUtil.java:503)
        at org.ballerinalang.test.util.BCompileUtil.compileOnJBallerina(BCompileUtil.java:730)
        at org.ballerinalang.test.util.BCompileUtil.compileOnJBallerina(BCompileUtil.java:624)
        at org.ballerinalang.test.util.BCompileUtil.compileOnJBallerina(BCompileUtil.java:612)
        at org.ballerinalang.test.util.BCompileUtil.compileOnJBallerina(BCompileUtil.java:605)
        at org.ballerinalang.test.util.BCompileUtil.compileOnJBallerina(BCompileUtil.java:691)
        at org.ballerinalang.test.util.BCompileUtil.compile(BCompileUtil.java:110)
        at org.ballerinalang.test.parser.ResilientParserTest.testResilientParsingModuleDeclaration(ResilientParserTest.java:62)
```

**Affected Versions:**
2.0.0-Preview1-M6

**Comment by SupunS** [comment]
This recovers and ands with a module level var decl. Then fails at the transormer due to https://github.com/ballerina-platform/ballerina-lang/issues/24878

**Comment by rdulmina** [comment]
Fixed in #26949
